client/state/current-user/reducer.js

```diff
@@ -82,9 +90,28 @@ export const capabilities = createReducer( {}, {
 	}
 }, capabilitiesSchema );
 
+export const gravatarStatus = createReducer( {}, {
+	[ GRAVATAR_UPLOAD_START ]: state => {
+		return Object.assign( {}, state, {
+			isUploading: true
+		} );
+	},
+	[ GRAVATAR_UPLOAD_SUCCESS ]: state => {
+		return Object.assign( {}, state, {
+			isUploading: false
+		} );
+	},
+	[ GRAVATAR_UPLOAD_FAILURE ]: state => {
+		return Object.assign( {}, state, {
+			isUploading: false
+		} );
+	},
+}, gravatarSchema );
```

Comment:
We generally never persist uploading state. We can get into a state where we never attempt to retry.